graph/cytoscape/cytoscape.go

```diff
@@ -107,16 +107,17 @@ func NewConfig(trafficMap graph.TrafficMap, o options.VendorOptions) (result Con
 	}
 
 	// sort nodes and edges for better json presentation (and predictable testing)
+	// kiali-1258 compound/isGroup/parent nodes must come before the child references
 	sort.Slice(nodes, func(i, j int) bool {
 		switch {
-		case nodes[i].Data.App < nodes[j].Data.App:
-			return true
-		case nodes[i].Data.App > nodes[j].Data.App:
-			return false
-		case nodes[i].Data.Version < nodes[j].Data.Version:
-			return true
-		case nodes[i].Data.Version > nodes[j].Data.Version:
-			return false
+		case nodes[i].Data.Namespace != nodes[j].Data.Namespace:
+			return nodes[i].Data.Namespace < nodes[j].Data.Namespace
+		case nodes[i].Data.IsGroup != nodes[j].Data.IsGroup:
+			return nodes[i].Data.IsGroup > nodes[j].Data.IsGroup
```

Comment:
should add a comment here to say ">" is on purpose (the others all use "<" so it looks like this is a typo and inconsistent)

Reply:
it's because the zero value for isGroup (string) is "".  In general we want to sort alphabetically so that we see "a" before "b".  But we want isGroup="version" before isGroup="" and "" < "version".